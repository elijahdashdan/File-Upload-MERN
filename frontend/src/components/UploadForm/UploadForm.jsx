import React, { useState } from 'react';
import { Row, Form, Image} from 'react-bootstrap';
import { fileUpload } from '../../data/api';
import UploadButton from '../UploadButton/UploadButton';


function UploadForm(props) {
    
    const [uploadFile, setUploadFile] = useState('');
    console.log(uploadFile)
    function uploadFileChange(event) {
        setUploadFile(event.target.files[0])
    }

    async function uploadFileClick() {

        if (uploadFile.length !== 0) {
            const formData = new FormData();
            formData.append('file', uploadFile);
            await fileUpload(formData);
            console.log(uploadFile);
            props.updateList();
        }
    }

    return (
        <div className="uploadForm">
            <Row>
                <Image src="cloud.png" fluid style={{ width: 150, height: 'auto' }} />
            </Row>
            <Row>
                <Form className="uploadForm">
                    <Form.Group controlId="formFileLg" className="mb-3">
                        <Form.Label>Select File to Upload</Form.Label>
                        <Form.Control onChange={(event) => uploadFileChange(event)} type="file" size="lg" />
                    </Form.Group>
                </Form>
                <UploadButton onClick = {uploadFileClick} />
            </Row>
        </div>

    )
}

export default UploadForm;