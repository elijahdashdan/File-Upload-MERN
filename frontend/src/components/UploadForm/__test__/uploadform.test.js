import React from 'react';
import ReactDOM from 'react-dom';
import UploadForm from '../UploadForm';
import renderer from 'react-test-renderer';

it("Renders without Crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<UploadForm />,div)
    ReactDOM.unmountComponentAtNode(div)
})

it('renders correctly', () => {
    const tree = renderer
      .create(<UploadForm />)
      .toJSON();
    expect(tree).toMatchSnapshot();
});