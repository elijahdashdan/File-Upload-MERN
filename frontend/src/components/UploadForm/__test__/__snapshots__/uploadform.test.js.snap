// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="uploadForm"
>
  <div
    className="row"
  >
    <img
      className="img-fluid"
      src="cloud.png"
      style={
        Object {
          "height": "auto",
          "width": 150,
        }
      }
    />
  </div>
  <div
    className="row"
  >
    <form
      className="uploadForm"
    >
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="formFileLg"
        >
          Select File to Upload
        </label>
        <input
          className="form-control form-control-lg"
          id="formFileLg"
          onChange={[Function]}
          type="file"
        />
      </div>
    </form>
    <div
      className="uploadButton"
    >
      <button
        className="btn btn-primary btn-lg"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Upload
      </button>
       
    </div>
  </div>
</div>
`;
